keps/sig-scheduling/5007-device-attach-before-pod-scheduled/README.md

```diff
@@ -0,0 +1,1043 @@
+
+In the DRA current implementation, fabric devices are attached after the scheduling decision, which leads to the following issues:
+
+Fabric devices may be contested by other clusters. In scenarios where attachment occurs after scheduling, there is a risk that the resource cannot be attached at the time of attachment, causing the container to remain in the "Container Creating" state.
+
+To address this issue, we propose a feature that allows the DRA scheduler plugin to wait for the device to be attached.
+
+## Motivation
+
+<!--
+This section is for explicitly listing the motivation, goals, and non-goals of
+this KEP.  Describe why the change is important and the benefits to users. The
+motivation section can optionally provide links to [experience reports] to
+demonstrate the interest in a KEP within the wider Kubernetes community.
+
+[experience reports]: https://github.com/golang/go/wiki/ExperienceReports
+-->
+
+As AI and ML become popular in container (K8s) environment, enormous computational resources are required more and more. On the other hand, efforts toward energy efficiency are also required for the realization of a sustainable society. It is expected to achieve the conflicting requirements that providing higher performance and reducing power consumption simultaneously. Recently, a new server architecture called Composable Disaggregated Infrastructure is emerged.
+
+In a traditional server, hardware resources such as CPUs, memory, and GPUs reside within the server. Composable Disaggregated Infrastructure decomposes these hardware resources and makes them available as resource pools. We can combine these resource by software definition so that we can create custom-made servers.
+
+Composable system is composed of resource pool and Composable Manager software. In Resource pool all components are connected to PCIe or CXL switches. Composable Manager controls the switches so as to create composed baremetals by software definition. It has Composable API and Operator or Kubernetes may call the API. Once composed baremetals are created user can install any operating system or container infrastructure.
+
+This flexibility extends further with the use of fabric devices. Fabric devices can be used by multiple Kubernetes clusters, not just a single one. Each cluster expose the device as a ResourceSlice, allowing for efficient utilization of the device.
+
+In this scenario, the ResourceSlice representing a same fabric device might be selected in multiple Kubernetes clusters simultaneously. If the attachment fails in one cluster, the pod will remain in a failed state in kubelet.
+
+By having the scheduler wait for the fabric device to be attached, we can reschedule the pod if the attachment fails. This approach is superior because it avoids unnecessary waiting and allows for immediate rescheduling.
```

Comment:
I am new to this area of CDI, but would appreciate if we can show how this para is more tightly related to the proposal ? For example does this make it easier to attach requested resource directly to the node where the pod is about to be scheduled or some other example like https://cdi.liqid.com/hubfs/Inference-aaS-WP_FNL.pdf